cmds/exp/dmidecode/dmidecode.go

```diff
@@ -60,19 +61,68 @@ func parseTypeFilter(typeStrings []string) (map[smbios.TableType]bool, error) {
 	return types, nil
 }
 
+func dumpBin(textOut io.Writer, entryData, tableData []byte, fileName string) *dmiDecodeError {
```

Comment:
Rather than passing in a `fileName`, can you pass in an `io.Writer` and make the caller responsible for opening/closing the file? This will make this function a bit more versatile. For example, `testDumpBin` could simply create a `bytes.Buffer` rather than creatin a temporary file.

Reply:
i thought about that but the thing is, i want to make sure entrypoint can be parsed and patched as needed first.
i don't want to create an empty file, then realize i can't actually write anything to it.
so, the test is a bit ugly, yes, but actual code behaves better in case of error.